store/migration/dev/LATEST.sql

```diff
@@ -21,11 +21,11 @@ CREATE TABLE principal (
     type TEXT NOT NULL CHECK (type IN ('END_USER', 'SYSTEM_BOT', 'SERVICE_ACCOUNT')),
     name TEXT NOT NULL,
     email TEXT NOT NULL,
-    password_hash TEXT NOT NULL
+    password_hash TEXT NOT NULL,
+    idp_id INTEGER REFERENCES idp (id),
+    idp_user_info JSONB DEFAULT '{}' CONSTRAINT principal_idp_id_idp_user_info_check CHECK ((idp_id IS NULL AND idp_user_info = '{}') OR (idp_id IS NOT NULL AND idp_user_info != '{}'))
```

Comment:
Don't feel like the check is necessary. Reduce the complexity.

Reply:
removed